File: 2015/day01/p2.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "../utils/utils.h"

int p2(char *input) {
    int floor = 0;
    int position = 0;
    // Return the first time Santa's position is in the
    // basement, or -1.
    for(int i = 0; i < strlen(input); i++) {
        position += 1;
        if(input[i] == '(') {
            floor += 1;
        } 
        else {
            floor -= 1;
        }
        if(floor == -1) {
            break;
        }
    }
    return position;
}

int main(void) {
    char *input = read_chars("in.txt");
    int res = p2(input);
    printf("res: %d\n", res);
    return 0;
}

File: 2015/01-06/01/p2.php
<?php
  // If character is (, go up one floor
  // If character is ), go down one floor
  // Return the first time Santa is in the basement
  $input = file_get_contents("in.txt");
  $floor = 0;
  $position = 0;  
  for($i = 0; $i < strlen($input); $i++) {
    $position++;
    if($input[$i] == '(') {
      $floor++;
    } else {
      $floor--;
    }
    if($floor == -1) {
      break;
    }
  }
  echo "p2: $position\n";
?>